src/OSPSuite.Core/Domain/Services/ParameterIdentifications/ParameterIdentificationRun.cs

```diff
@@ -229,10 +235,14 @@ private IReadOnlyList<SimulationRunResults> runSimulations()
 
       private void updateParameterValues(IReadOnlyList<OptimizedParameterValue> values)
       {
-         if (_parameterIdentification.AllIdentificationParameters.Count != values.Count)
-            throw new ArgumentException(Error.IdentificationParametersAndValuesDoNotTheSameLength(_parameterIdentification.AllIdentificationParameters.Count, values.Count));
+         if (_variableParameters.Count != values.Count)
+            throw new ArgumentException(Error.IdentificationParametersAndValuesDoNotTheSameLength(_variableParameters.Count, values.Count));
 
-         _parameterIdentification.AllIdentificationParameters.Each((x, i) => addParameterValue(x, values[i].Value));
+         //Update variable parameters according to optimized parameter values
+         _variableParameters.Each((x, i) => addParameterValue(x, values[i].Value));
+
+         //Preserve fixed parameter values
+         _fixedParameters.Each((x, i) => addParameterValue(x, x.StartValue));
```

Comment:
ok. so you mean in this method here
```
  private ISimModelBatch createSimModelBatch(ISimulation simulation)
      {
         var simModelBatch = _simModelBatchFactory.Create();
         var modelCoreSimulation = _modelCoreSimulationMapper.MapFrom(simulation, shouldCloneModel:true);
         _timeGridUpdater.UpdateSimulationTimeGrid(modelCoreSimulation, _parameterIdentification.Configuration.RemoveLLOQMode, _parameterIdentification.AllDataRepositoryMappedFor(simulation));
         _outputSelectionUpdater.UpdateOutputsIn(modelCoreSimulation, _parameterIdentification.AllOutputsMappedFor(simulation));
         simModelBatch.InitializeWith(modelCoreSimulation, _parameterIdentification.PathOfOptimizedParameterBelongingTo(simulation), simulationResultsName: Captions.ParameterIdentification.SimulationResultsForRun(RunResult.Index));
         return simModelBatch;
      }
```

after
```
  var modelCoreSimulation = _modelCoreSimulationMapper.MapFrom(simulation, shouldCloneModel:true);
```
we could update the fixed value? yeah that sounds like a good idea. Then I could remove all fixed parameters from the initialization of SimModel correct